.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,53 @@
+name: Build and Publish
+
+on:
+  push:
+    branches:
+      - main
+      - release/*
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```

Comment:
I would use a specific version to make sure the build is reproducible. I see other OTEL projects use ubuntu-20.04, e.g. [here](https://github.com/open-telemetry/opentelemetry-proto-java/blob/main/.github/workflows/main-build.yml)